ibm_cloud_sdk_core/base_service.py

```diff
@@ -314,6 +323,38 @@ def send(self, request: requests.Request, **kwargs) -> DetailedResponse:
         try:
             response = self.http_client.request(**request, cookies=self.jar, **kwargs)
 
+            # Handle HTTP redirects.
+            redirects_count = 0
+            # Check if the response is a redirect to another host.
+            while response.is_redirect and response.next is not None and redirects_count < MAX_REDIRECTS:
+                redirects_count += 1
+
+                # urllib3 has already prepared a request that can almost be used as-is.
+                next_request = response.next
+
+                # If both the original and the redirected URL are under the `.cloud.ibm.com` domain,
+                # copy the safe headers that are used for authentication purposes,
+                if self.service_url.endswith('.cloud.ibm.com') and urlparse(next_request.url).netloc.endswith(
```

Comment:
`self.service_url.endswith('.cloud.ibm.com')` will most definitely not work in the general case. To see why, consider a service url like `https://myhost.ibm.com/api` or `https://myhost.ibm.com:8080`.  I think you'll need to parse `self.service_url` and extract the hostname from it.

Reply:
Yeah, that's a good point and I already parse the url in the second part of the expression... :)